src/CoreFx.Private.TestUtilities/src/System/IO/FileCleanupTestBase.cs

```diff
@@ -84,7 +85,37 @@ protected virtual void Dispose(bool disposing)
         /// <param name="index">An optional index value to use as a suffix on the file name.  Typically a loop index.</param>
         /// <param name="memberName">The member name of the function calling this method.</param>
         /// <param name="lineNumber">The line number of the function calling this method.</param>
-        protected string GetTestFileName(int? index = null, [CallerMemberName] string memberName = null, [CallerLineNumber] int lineNumber = 0) =>
+        protected string GetTestFileName(int? index = null, [CallerMemberName] string memberName = null, [CallerLineNumber] int lineNumber = 0)
+        {
+            string testFileName = GenerateTestFileName(index, memberName, lineNumber);
+            string testFilePath = Path.Combine(TestDirectory, testFileName);
+
+            int excessLength = testFilePath.Length - 255;
+
+            if (excessLength > 0)
+            {
+                // The path will be too long for Windows -- can we
+                // trim memberName to fix it?
+                if (excessLength < memberName.Length + "...".Length)
+                {
+                    // Take a chunk out of the middle as perhaps it's the least interesting part of the name
+                    memberName = memberName.Substring(0, memberName.Length / 2 - excessLength / 2) + "..." + memberName.Substring(memberName.Length / 2 + excessLength / 2);
+
+                    testFileName = GenerateTestFileName(index, memberName, lineNumber);
+                    testFilePath = Path.Combine(TestDirectory, testFileName);
+                }
+                else
+                {
+                    throw new IOException($"Test method name '{memberName}' is too long to make a reasonable test directory path; use a shorter name");
```

Comment:
instead of throwing why not just create a random 8 character Guid and use that as the testFileName? So that we don't see random failures because people renamed the tests

Reply:
Good idea, fixed.